Bucketeer2/Sources/Internal/Scheduler/EvaluationForegroundTask.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+
+final class EvaluationForegroundTask: ScheduledTask {
+    private weak var component: Component?
+    private let queue: DispatchQueue
+    private var poller: Poller?
+    private var retryPollingInterval: Int64
+    private var maxRetryCount: Int
+
+    private var retryCount: Int = 0
+
+    init(component: Component,
+         queue: DispatchQueue,
+         retryPollingInterval: Int64 = Constant.RETRY_POLLING_INTERVAL,
+         maxRetryCount: Int = Constant.MAX_RETRY_COUNT) {
+
+        self.component = component
+        self.queue = queue
+        self.retryPollingInterval = retryPollingInterval
+        self.maxRetryCount = maxRetryCount
+    }
+
+    private func reschedule(isRetrying: Bool) {
+        self.stop()
+        guard let component = component else { return }
+        self.poller = .init(
+            intervalMillis: isRetrying ? retryPollingInterval : component.config.eventsFlushInterval,
+            queue: queue,
+            logger: component.config.logger,
+            handler: { [weak self] _ in
+                self?.fetchEvaluations()
+            }
+        )
+        poller?.start()
+    }
+
+    func start() {
+        reschedule(isRetrying: false)
+    }
+
+    func stop() {
+        poller?.stop()
+        poller = nil
+    }
+
+    private func fetchEvaluations() {
```

Comment:
This code which fetches the evaluations and report metrics, will be used in other places.
We can avoid duplicating this implementation making a static function and reusing it when needed.

You can use this as a reference:
- [fetchEvaluationsSync](https://github.com/bucketeer-io/android-client-sdk/blob/b6c05e4d5d065eec905f242addb1a1ed3543a256/bucketeer/src/main/kotlin/io/bucketeer/sdk/android/BKTClientImpl.kt#L179-L211)

Placing being used:
https://github.com/bucketeer-io/android-client-sdk/search?q=fetchEvaluationsSync

Reply:
Yes, it is.
Thanks.